paddle/fluid/framework/executor.cc

```diff
@@ -183,6 +183,9 @@ void Executor::Run(const ProgramDesc& pdesc, Scope* scope, int block_id,
                    bool force_disable_gc, bool keep_kid_scopes) {
   platform::RecordBlock b(block_id);
   if (FLAGS_use_mkldnn) EnableMKLDNN(pdesc);
+#ifdef PADDLE_WITH_MKLDNN
+  platform::AttachPointerHashToMKLDNNKey(this, place_);
```

Comment:
Please guard the call with if(FLAGS_use_mkldnn)

Reply:
You may have oneDNN ops executed without FLAGS_use_mkldnn. This is what happens in Unit tests for example.

Reply:
> Please guard the call with if(FLAGS_use_mkldnn)

@raindrops2sea and I agree with this. Since `AttachPointerHashToMKLDNNKey` may have some cost for other models.

> You may have oneDNN ops executed without FLAGS_use_mkldnn. This is what happens in Unit tests for example.

The PR description said "Due to huge number of modifies, unit tests will come in next PR". Is the same thing?

Reply:
@jczaja Could you guard the call with if(FLAGS_use_mkldnn)?

Reply:
@luotao1 No. We cannot do it. FLAGS_use_mkldnn makes ops that are having attribute: use_mkldnn to have it set to True. But you can set those attributes manualy (unit tests are doing it) . For situation when use_mkldnn attribute is set manually (not via FLAGS_use_mkldnn) we still want to call AttachPointerHashToMKLDNNKey even that FLAGS_use_mkldnn is not set. So We cannot guard this call with if(FLAGS_use_mkldnn)

Reply:
@jczaja  Got it. But how to avoid this problem:
AttachPointerHashToMKLDNNKey may have some cost for other models which do not use mkldnn?

Reply:
@luotao1 , @raindrops2sea I analyzed it more, and actually with some other fixes (introduced earlier in this PR) changes in Executor.cc are not needed e.g. AttachPointerHashToMKLDNNKey is part of Executor::EnableMKLDNN()